pkg/agent/nodeportlocal/npl_agent_test.go

```diff
@@ -293,12 +301,17 @@ func (t *testData) checkAnnotationValue(value string, podPort ...int) []nplk8s.N
 	var nplValue []nplk8s.NPLAnnotation
 	err := json.Unmarshal([]byte(value), &nplValue)
 	require.NoError(t, err, "Error when unmarshalling NPL annotation")
-
-	require.Equal(t, len(podPort), len(nplValue))
-	for idx, port := range podPort {
-		assert.Equal(t, defaultHostIP, nplValue[idx].NodeIP)
-		assert.Equal(t, port, nplValue[idx].PodPort)
+	podPortSet := sets.NewInt(podPort...)
+	require.Equal(t, podPortSet.Len(), len(nplValue))
+	for port := range podPortSet {
+		for _, npl := range nplValue {
+			if port == npl.PodPort {
+				podPortSet.Delete(port)
+				assert.Equal(t, defaultHostIP, npl.NodeIP)
+			}
+		}
 	}
+	require.Empty(t, podPortSet.Len())
```

Comment:
there is no protocol check in this function?

I would require something like this for readability, even if it's a bit more verbose:
```go
type annotation struct {
    podPort int
    protocols []string
    hostIP string
}

type NPLAnntotationChecker struct {
    expectedAnnotations []annotation
}

func (c *NPLAnntotationChecker) addPort(podPort int, protocols ...string) *NPLAnntotationChecker {
    a := annotation{podPort: podPort, hostIP: defaultHostIP}
    a.protocols = append(a.protocols, protocols...)
    sort.Sort(a.protocols)
    c.expectedAnnotations = append(c.expectedAnnotations, a)
}

func (c *NPLAnntotationChecker) matches(t *testing.T, value string) {
    var nplValue []nplk8s.NPLAnnotation
    err := json.Unmarshal([]byte(value), &nplValue)
    require.NoError(t, err, "Error when unmarshalling NPL annotation")
    annotations := []annotation{}
    for _, nplAnnotation := range nplValue {
        a := annotation{podPort: nplAnnotation.PodPort, protocols: nplAnnotation.Protocols, hostIP: nplAnnotation.NodeIP}
        sort.Sort(a.protocols)
        annotations = append(annotations, a)
    }
    assert.ElementsMatch(t, annotations, c.expectedAnnotations)
}
```

I think it will make tests easier to read and also facilitate debugging.